azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/b2c/AADB2CFilterSignUpOrInHandler.java

```diff
@@ -25,42 +32,99 @@
 public class AADB2CFilterSignUpOrInHandler extends AbstractAADB2CFilterScenarioHandler
         implements AADB2CFilterScenarioHandler {
 
+    private final RedirectStrategy redirectStrategy = new DefaultRedirectStrategy();
+
     /**
      * Mapping configuration URL to ${@link AADB2CJWTProcessor}.
      */
-    private final ConcurrentMap<String, AADB2CJWTProcessor> urlToJwtParser = new ConcurrentHashMap<>();
+    private static final ConcurrentMap<String, AADB2CJWTProcessor> URL_TO_JWT_PARSER = new ConcurrentHashMap<>();
 
     /**
      * Get cached instance of ${@link AADB2CJWTProcessor} from given url, or create new one.
      *
      * @param url        of the configuration.
      * @param properties of ${@link AADB2CProperties}.
      * @return the instance of ${@link AADB2CJWTProcessor}.
-     * @throws AADB2CAuthenticationException when failed to create ${@link AADB2CJWTProcessor} instance.
      */
-    private AADB2CJWTProcessor getAADB2CJwtParser(@URL String url, @NonNull AADB2CProperties properties)
-            throws AADB2CAuthenticationException {
-        urlToJwtParser.putIfAbsent(url, new AADB2CJWTProcessor(url, properties));
+    private AADB2CJWTProcessor getAADB2CJwtProcessor(@URL String url, @NonNull AADB2CProperties properties) {
+        // TODO(pan): url should exclude state and nonce UUID part.
+        URL_TO_JWT_PARSER.putIfAbsent(url, new AADB2CJWTProcessor(url, properties));
```

Comment:
`computeIfAbsent`? We should delay creating new instance.

Reply:
yes, should be computeIfAbsent.